plugins/BEdita/API/src/Utility/JsonApi.php

```diff
@@ -77,6 +76,44 @@ protected static function extractType($item, $type)
     }
 
     /**
+     * Build URL to be used in `links` object.
+     *
+     * @param string $name Route name.
+     * @param string $endpoint Endpoint.
+     * @param string $type Resource type.
+     * @param array $options Additional options.
+     * @return string|null
+     */
+    protected static function buildUrl($name, $endpoint, $type, array $options = [])
+    {
+        $url = null;
+
+        $options['_name'] = sprintf('api:resources:%s', $name);
+        $options['controller'] = $type;
+
+        try {
+            $url = Router::url($options, true);
+        } catch (MissingRouteException $e) {
+            $options['_name'] = sprintf('api:%s:%s', $endpoint, $name);
+            unset($options['controller']);
+
+            try {
+                $url = Router::url($options, true);
+            } catch (MissingRouteException $e) {
+                $options['object_type'] = $type;
+
+                try {
+                    $url = Router::url($options, true);
+                } catch (MissingRouteException $e) {
+                    // Do not halt if route is missing.
```

Comment:
Surely it's a weird scenario. I tried re-throwing the exception, and the whole test suite does pass, so we can probably assume that in this case something is going wrong, and the developer wants to be informed about that.